staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -326,6 +326,8 @@ type CustomResourceDefinitionCondition struct {
 // CustomResourceDefinitionStatus indicates the state of the CustomResourceDefinition
 type CustomResourceDefinitionStatus struct {
 	// Conditions indicate state for particular aspects of a CustomResourceDefinition
+	// +listType=map
+	// +listMapKey=type
```

Comment:
Does this change belong in this PR?

Reply:
The way that we test status wiping is by applying a change with a new dummy condition in the .status.conditions.

For CRDs, the CRD controller updates the conditions as part of the test after the initial create is called but before the status updating Apply call is made. Without the change, the conditions list is atomic and so the apply that adds a new condition will always fail with a conflict. 

There are no other fields on a CRD status that the apply can update instead, and it is incorrect anyways for the conditions to be atomic, so we changed it here. After we get approval, when we go to squash commits, it would make sense to have this as a separate commit.

Reply:
I agree that it's correct it's just surprising to combine it with this change. Does this have any compatibility implications?

Reply:
Joe has worked specifically on making this possible.